### PR TITLE
Centralize design token schema

### PR DESCRIPTION
**Pull request by connor-baer**
## Purpose

The list of CSS custom properties is duplicated in several places in the monorepo (ESLint plugin, BaseStyles, Storybook). This PR takes the first steps to centralize the token names in the `@sumup/design-token` package.

## Approach and changes

- Export the token schema from `@sumup/design-tokens` to act as the source of truth for validating design tokens. It replaces the hard-coded values in the ESLint plugin and Storybook theme docs.

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: package-lock.json
Comment (connor-baer) [inline package-lock.json:1]: This file had conflicts when rebasing on `main` and apparently npm decided to migrate to upgrade to lockfile version 3 while resolving them 🤷‍♂️

Comment (connor-baer) [reply package-lock.json:1]: It appears it also upgraded all dependencies and broke CI, so I reverted the changes.

File: docs/features/1-theme.mdx [hunk 1]
```diff
@@ -32,170 +34,15 @@ To make it easier to work with custom properties, we recommend using the [`@sumu
...
-    '--cui-bg-promo-hovered',
-    '--cui-bg-promo-pressed',
-    '--cui-bg-promo-disabled',
-    '--cui-bg-promo-strong',
-    '--cui-bg-promo-strong-hovered',
-    '--cui-bg-promo-strong-pressed',
-    '--cui-bg-promo-strong-disabled',
-  ]}
-/>
+<ColorsTable customProperties={filterCustomProperties(schema, 'bg')} />
```
Comment (robinmetral) [inline docs/features/1-theme.mdx:37]: Nice!

**Review by robinmetral — COMMENTED** [review]
Super cool to avoid duplicating these! ✨

File: package-lock.json [hunk 2]
```diff
@@ -30096,8 +30096,12 @@
         "@typescript-eslint/utils": "^5.59.2"
       },
       "devDependencies": {
+        "@sumup/design-tokens": "^5.0.0",
```
[the remark this answers is not in the document]
Comment (connor-baer) [reply package-lock.json:30099]: Yes, peer dependencies should also be specified as dev dependencies for local development.

File: packages/design-tokens/themes/schema.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { TokenName, TokenType } from '../types';
+
+export const schema: { name: TokenName; type: TokenType }[] = [
+  /* Neutral backgrounds */
+  { name: '--cui-bg-normal', type: 'color' },
```
[the remark this answers is not in the document]
Comment (connor-baer) [reply packages/design-tokens/themes/schema.ts:20]: Yes. The `type` property is useful to filter the properties or to select an appropriate transformer without having to manually parse the name.

**Pull request merged**